samples/TagHelperSample.Web/TagHelpers/ConditionalCommentTagHelper.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+
+namespace TagHelperSample.Web
+{
+    [TargetElement("iecondition")]
+    public class ConditionalCommentTagHelper : TagHelper
+    {
+        [HtmlAttributeName("mode")]
+        public string Mode { get; set; }
+
+        [HtmlAttributeName("condition")]
+        public string Condition { get; set; }
+
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            output.TagName = null;
+            string startCondition = "";
+            string endCondition = "";
+            if (Mode == "downlevelHidden")
+            {
+                startCondition = $"<!--[if {Condition}]>";
+                endCondition = "<![endif]-->";
+
+            }
+            else if (Mode == "downlevelRevealed")
+            {
+                startCondition = $"<![if {Condition}]>";
```

Comment:
Can't HTML encode because `&` is a special character in the [syntax](https://msdn.microsoft.com/en-us/library/ms537512.aspx). So there's no secure way to do this.

But still not great to do the extra concatenation. Updated my previous comment...